### PR TITLE
[FEAT]: Added true balance viewing 💰

### PR DESCRIPTION
**Pull request by Pranav-yadav**
# Description

Added feature to view the **True Balance** of the player. Currently the balance is shown by rounding up to 2 dec places which sometimes confuses player when they want to Buy some resources in game.
So, in `Balance.tsx` added **`onMouseEnter`** and **`onMouseLeave`** to span which shows rounded balance such that, _when user **hovers** over the Balance Area it gets **expanded** and shows True Balance in **smaller font**- (for better visibility on mobile devices)._ Ref. screenshots.

#### Why this functionality?
=> Players won't be confused while Buying Or Selling resources also they'll have more grasp over their **True Balance**
Below screenshots show some scenarios where knowing True Balance seems necessary.

<details><summary><b>Screenshots</b></summary>
<ul>
<li>Before

</li>
<li>After

</li>
<li>Before

</li>
<li>After

</ul>

<li>Before

</li>
<li>After

</ul>
</details>

Fixes --

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

yarn test

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/hud/components/Balance.tsx [hunk 1]
```diff
@@ -11,14 +11,23 @@ import Decimal from "decimal.js-light";
...
 
   return (
     <InnerPanel className="fixed top-2 right-2 z-50 flex items-center shadow-lg">
       <img src={token} className="w-8 img-highlight" />
-      <span className="text-white text-sm text-shadow ml-2">
-        {game.context.state.balance
-          .toDecimalPlaces(2, Decimal.ROUND_DOWN)
-          .toString()}
+      <span
+        className="text-white text-sm text-shadow ml-2"
```
Comment (adamhannigan) [inline src/features/hud/components/Balance.tsx:20]: Consider adding the `cursor-pointer` class to show a cursor on hover

**Review by adamhannigan — APPROVED** [review]
Looks good to me 💪 

I've just small a small UX suggestion.

Also, just a heads up this is how it looks on mobile when you click it. It minimizes once you click elsewhere

Screen Shot 2022-01-25 at 4 24 44 pm

**Comment by Pranav-yadav** [comment]
@adamhannigan required changes are done. Do review and merge :)

**Review by adamhannigan — APPROVED** [review]
LGTM 💪

**Pull request merged**